torch/fx/subgraph_rewriter.py

```diff
@@ -190,7 +194,16 @@ def pattern_is_contained(nodes_map : Dict[Node, Node]) -> bool:
                 lookup: Dict[Node, Node] = {v : k for k, v
                                             in nodes_map.items()}
                 for n in lookup.keys():
-                    if n.op == "placeholder" or lookup[n].op == "output":
+                    # Nodes that can "leak":
+                    #   - Placeholders (by definition)
+                    #   - Pattern output (acts as a container)
+                    #   - Result contained by pattern output (what we'll
+                    #     hook in to the new Graph, thus what we'll
+                    #     potentially use in other areas of the Graph as
+                    #     an input Node)
+                    if (n.op == "placeholder" or lookup[n].op == "output"
+                        or (len(lookup[n].users) == 1 and
+                            list(lookup[n].users.keys())[0].op == "output")):
```

Comment:
This is pretty hard to read. I suggest creating separate `if XXX: continue` branches for each of these `or`'ed conditions, and probably move in the comment for each of those specific situations into the branches